From 44e0647230b22cd136ae2fb252ce90eed4751970 Mon Sep 17 00:00:00 2001
From: Alex Kiernan <alex.kiernan@gmail.com>
Date: Wed, 4 Sep 2019 17:29:15 +0100
Subject: [PATCH 1/2] Always enable trivial-httpd for tests

When running tests we always need ostree-trivial-httpd, so enable it
unconditionally

Signed-off-by: Alex Kiernan <alex.kiernan@gmail.com>
Upstream-Status: Submitted [https://github.com/ostreedev/ostree/pull/1912]
---
 Makefile.am        | 1 +
 ci/build.sh        | 5 +++++
 ci/travis-build.sh | 1 +
 3 files changed, 7 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index cd04a055045a..673dbf88de55 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -39,6 +39,7 @@ AM_DISTCHECK_CONFIGURE_FLAGS += \
 	--enable-gtk-doc \
 	--enable-man \
 	--disable-maintainer-mode \
+	--enable-trivial-httpd-cmdline \
 	$(NULL)
 
 GITIGNOREFILES = aclocal.m4 build-aux/ buildutil/*.m4 config.h.in gtk-doc.make
diff --git a/ci/build.sh b/ci/build.sh
index 0901507498f2..806af050b780 100755
--- a/ci/build.sh
+++ b/ci/build.sh
@@ -22,6 +22,11 @@ case "${CONFIGOPTS:-}" in
         fi
         ;;
 esac
+# unless libsoup is disabled, enable trivial-httpd for the tests
+case "${CONFIGOPTS:-}" in
+    *--without-soup*) ;;
+    *) CONFIGOPTS="${CONFIGOPTS:-} --enable-trivial-httpd-cmdline" ;;
+esac
 
 # always fail on warnings; https://github.com/ostreedev/ostree/pull/971
 # NB: this disables the default set of flags from configure.ac
diff --git a/ci/travis-build.sh b/ci/travis-build.sh
index 3fd969bdced6..7c85313a2d32 100755
--- a/ci/travis-build.sh
+++ b/ci/travis-build.sh
@@ -85,6 +85,7 @@ make="make -j${ci_parallel} V=1 VERBOSE=1"
 
 ../configure \
     --enable-always-build-tests \
+    --enable-trivial-httpd-cmdline \
     ${ci_configopts}
     "$@"
 
-- 
2.17.1

